src/lib.rs

```diff
@@ -883,7 +889,12 @@ impl<TRepoTypes: RepoTypes> Future for IpfsFuture<TRepoTypes> {
                         ret.send(Ok(addrs)).ok();
                     }
                     IpfsEvent::Listeners(ret) => {
-                        let listeners = Swarm::listeners(&self.swarm).cloned().collect();
+                        let mut listeners = Swarm::listeners(&self.swarm)
+                            .cloned()
+                            .collect::<Vec<Multiaddr>>();
+                        for addr in &mut listeners {
+                            addr.push(Protocol::P2p(self.swarm.ipfs().peer_id().into()));
+                        }
```

Comment:
Why put the p2p/peerid here? go-ipfs 0.6.0 does not add the peerid to these responses.

Reply:
We can not add it here - it will just require a few tests to be adjusted, but other than that it's not needed.